hxinsight-extension/src/main/java/org/alfresco/hxi_connector/hxi_extension/rest/api/KnowledgeRetrievalEntityResource.java

```diff
@@ -0,0 +1,48 @@
+/*-
+ * #%L
+ * Alfresco HX Insight Connector
+ * %%
+ * Copyright (C) 2023 - 2024 Alfresco Software Limited
+ * %%
+ * This file is part of the Alfresco software.
+ * If the software was purchased under a paid Alfresco license, the terms of
+ * the paid license agreement will prevail.  Otherwise, the software is
+ * provided under the following open source license terms:
+ *
+ * Alfresco is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Alfresco is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Alfresco. If not, see <http://www.gnu.org/licenses/>.
+ * #L%
+ */
+package org.alfresco.hxi_connector.hxi_extension.rest.api;
+
+import java.util.Set;
+
+import lombok.RequiredArgsConstructor;
+
+import org.alfresco.rest.framework.resource.EntityResource;
+import org.alfresco.rest.framework.resource.actions.interfaces.EntityResourceAction;
+import org.alfresco.rest.framework.resource.parameters.CollectionWithPagingInfo;
+import org.alfresco.rest.framework.resource.parameters.Parameters;
+
+@RequiredArgsConstructor
+@EntityResource(name = "knowledge-retrieval", title = "Hyland Knowledge Retrieval Url")
+public class KnowledgeRetrievalEntityResource implements EntityResourceAction.Read<String>
+{
+    private final String knowledgeRetrievalUrl;
+
+    @Override
+    public CollectionWithPagingInfo<String> readAll(Parameters params)
```

Comment:
As discussed - let's return an config object here without wrapping it with `CollectionWithPagingInfo`

Reply:
This will likely lead to changing implementation to extend `RelationshipResourceAction.ReadById` instead of `RelationshipResourceAction.Read` which will require introducing an `id` path parameter.
There are places where we used `-default-` as this `id` in such situations.

Reply:
good point - changed